src/vs/workbench/contrib/terminal/common/terminalEnvironment.ts

```diff
@@ -190,17 +191,29 @@ export function getDefaultShell(
 		executable = executable.replace(/\//g, '\\');
 	}
 
+	if (configurationResolverService) {
+		executable = configurationResolverService.resolve(undefined, executable);
+	}
+
 	return executable;
 }
 
 export function getDefaultShellArgs(
 	fetchSetting: (key: string) => { user: string | string[] | undefined, value: string | string[] | undefined, default: string | string[] | undefined },
 	isWorkspaceShellAllowed: boolean,
-	platformOverride: platform.Platform = platform.platform
+	configurationResolverService: IConfigurationResolverService | undefined,
+	platformOverride: platform.Platform = platform.platform,
 ): string[] {
 	const platformKey = platformOverride === platform.Platform.Windows ? 'windows' : platformOverride === platform.Platform.Mac ? 'osx' : 'linux';
 	const shellArgsConfigValue = fetchSetting(`terminal.integrated.shellArgs.${platformKey}`);
-	const args = (isWorkspaceShellAllowed ? <string[]>shellArgsConfigValue.value : <string[]>shellArgsConfigValue.user) || <string[]>shellArgsConfigValue.default;
+	let args = (isWorkspaceShellAllowed ? <string[]>shellArgsConfigValue.value : <string[]>shellArgsConfigValue.user) || <string[]>shellArgsConfigValue.default;
+	if (configurationResolverService) {
+		const resolvedArgs: string[] = [];
```

Comment:
Just replace `args` in place instead of using `let` and creating a new array

Reply:
What do you mean by replacing it in place? If I try to loop through and resolve like `args[I] = resolved(args[I])` I get runtime errors and I'm unsure why.

Reply:
Ah ok, they come from the config service frozen I guess.